mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -192,10 +193,11 @@ public void setCurrentKillstreaks(Transaction transaction, int amount) {
         if (amount <= 0) {
             throw new IllegalArgumentException("amount must be positive");
         }
-        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
 
-        statisticsRecord.setCurrentKillstreak(amount);
-        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+        transaction.getProperty(DSLContext.class).update(KITPVP_STATISTICS)
+                .set(KITPVP_STATISTICS.HIGHEST_KILLSTREAK,KITPVP_STATISTICS.HIGHEST_KILLSTREAK.greatest(amount))
```

Comment:
Is this `greatest` method really deprecated? If so, an alternative should be used

Reply:
I looked for an alternative. the static method greatest() is not deprecated (not field.greatest()) but set does not seem to accept the field returned from greatest(s,d)

Reply:
What types does the static method take? What types are you trying to give it, or how do you think you could give it the types it wants? What type does the method return?

Reply:
The static method takes either a T and a T, or it takes a Field<T> and a Field<?>...
I am giving it the TableField KITPVP_STATISTICS.HIGHEST_KILLSTREAK and an integer
It returns a Field, but for some reason set doesn't want to accept it

Reply:
What's the some reason that the set method does not accept it?

Since the method takes either a T and T or a Field and Field, you either need to convert your TableField to an Integer, or your Integer to a TableField. That then raises the question whether jooq has some kind of "literal" method which returns a Field<T> from a T. I would guess that such a method exists.